src/main/java/com/iota/iri/storage/Tangle.java

```diff
@@ -82,13 +83,38 @@ public void shutdown() throws Exception {
      * @see PersistenceProvider#get(Class, Indexable)
      */
     public Persistable load(Class<?> model, Indexable index) throws Exception {
-            Persistable out = null;
-            for(PersistenceProvider provider: this.persistenceProviders) {
-                if((out = provider.get(model, index)) != null) {
-                    break;
+        LinkedList<Persistable> outlist = new LinkedList<>();
+        for (PersistenceProvider provider : this.persistenceProviders) {
+            Persistable result = provider.get(model, index);
+
+            if (result != null && result.exists()) {
+                if (result.merge()) {
+
+                    outlist.add(result);
+                } else {
+                    // If it is a non-mergeable result then there is no need to ask another provider again.
+                    // return immediately
+                    return result;
                 }
             }
-            return out;
+        }
+        Persistable p = outlist.stream().reduce(null, (a, b) -> {
+            if (a == null) {
+                return b;
+            }
+            try {
+                return a.mergeInto(b);
+            }catch (OperationNotSupportedException e){
+                log.error("Error merging data: ", e);
+                log.error("Error merging data: ", e);
+                return null;
+            }
+        });
+
+        if (p == null) {
+            p = (Persistable) model.newInstance();
```

Comment:
We used to return null in this case, is there a reason you need an empty instance?

Reply:
Actually there was never null returned here. Because persistence provider.get never returns null. It is the reason why this change is needed. But that code is in there for backward compatibility. Could go with a comment though.

Reply:
Aha, yea youre right. If you do the suggestion below, it will never be null anyways, so the check can be removed

Reply:
Well I think it should be possible to be null actually but the check is there 'outside' for the reason it looks 'wrong' maybe not the right way to do it, but neither is including class instantiation just because. : 
https://github.com/iotaledger/iri/blob/6769054eb05b07639d714d67eeb34d022e5fe1bd/src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java#L154-L155

But really the right way would be to let the persistence provider return null if it didn't find anything and fix everything it breaks. But that will be a different issue then.

I would advise for maybe a new issue describing it and add it to the comments on why this is necessary.

Reply:
https://github.com/iotaledger/iri/issues/1591

I chose to create an empty immutable constant object just to avoid NullPointerExceptions